tools/clang/unittests/HLSLExec/ShaderOpArith.xml

```diff
@@ -4215,6 +4215,31 @@ void MSMain(uint GID : SV_GroupIndex,
         }
         #endif
 
+        #ifdef FUNC_WAVE_MATCH
+        void TestWaveMatch(vector<TYPE, NUM> Vector)
+        {
+            if(WaveGetLaneIndex() == 0)
+            {
+              if(Vector[0] == (TYPE)0)
+                Vector[0] = (TYPE) 1;
+              else if(Vector[0] == (TYPE)1)
+                Vector[0] = (TYPE) 0;
+              else
+                Vector[0] = (TYPE) 1;
+            }
+            uint4 result = WaveMatch(Vector);
+            uint index = WaveGetLaneIndex() * 4;
+            if(index < NUM)
```

Comment:
I'm not sure what the intent of using NUM was here. 
NUM describes the number of elements in the input Vector, which isn't related to the number of lanes in the wave.
I think what you will want to do is store the mask from each lane. That is, your output should have WaveGetLaneCount() elements. With each element being a uint4. And in this test case you would expect to have two unique masks. With the first mask being for the first group in lane 0 only. And all other lanes in another group (mask).

That would greatly simplify your logic for computing expected values as well.